presto-main/src/test/java/com/facebook/presto/sql/planner/TestLogicalPlanner.java

```diff
@@ -1016,6 +1017,66 @@ public void testEqualityInference()
                                                 tableScan("partsupp", ImmutableMap.of("p_suppkey", "suppkey", "p_partkey", "partkey", "p_comment", "comment")))))));
     }
 
+    @Test
+    public void testEmptyJoins()
+    {
```

Comment:
Good to add more than one join with each (first, middle, last) being empty and the result.

Reply:
Done for 3 way joins and more complex cases of inputs to joins like aggregations.